metadata_mapper/lambda_function.py

```diff
@@ -49,34 +53,30 @@ def parse_enrichment_url(enrichment_url):
 # {"collection_id": 26098, "mapper_type": "nuxeo", "page_filename": 2}
 # AWS Lambda entry point
 def map_page(payload, context):
-    if settings.LOCAL_RUN:
+    if settings.LOCAL_RUN and not isinstance(payload, dict):
         payload = json.loads(payload)
 
     vernacular_reader = import_vernacular_reader(payload.get('mapper_type'))
     source_vernacular = vernacular_reader(payload)
     api_resp = source_vernacular.get_api_response()
-    source_metadata_records = source_vernacular.parse(api_resp)
+    mapped_records = source_vernacular.parse(api_resp)
     collection = payload.get('collection', {})
 
-    for enrichment_url in collection.get('rikolti__pre_mapping'):
+    for enrichment_url in collection.get('rikolti__pre_mapping', []):
         enrichment_func, kwargs = parse_enrichment_url(enrichment_url)
         if not enrichment_func and settings.SKIP_UNDEFINED_ENRICHMENTS:
             continue
         logging.debug(
             f"[{collection['id']}]: running enrichment: {enrichment_func} "
             f"for page {payload['page_filename']} with kwargs: {kwargs}")
-        source_metadata_records = [
+        mapped_records = [
             record.enrich(enrichment_func, **kwargs)
-            for record in source_metadata_records
+            for record in mapped_records
         ]
 
-    mapped_records = [record.to_UCLDC() for record in source_metadata_records]
     writer = UCLDCWriter(payload)
-    if settings.DATA_DEST == 'local':
```

Comment:
We are interested in adding the capacity to see records directly after the mapping step, but before the bulk of the enrichments (excepting those "pre enrichments") have been run. I think this is largely for debug purposes and probably deserves some sort of flag.

Reply:
Reverted